packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_token_mint_token.test.ts

```diff
@@ -0,0 +1,364 @@
+  })
+
+  it('should throw an error if global mint limit exceeds', async () => {
+    await setupGovs()
+
+    // Hit global daily mint limit
+    await alice.rpc.token.mintTokens({ amounts: [`40.0000000@${symbolBTC}`] })
+    await alice.generate(1)
+    await tGroup.waitForSync()
+
+    const script = await bProviders.elliptic.script()
+    const tokenMint: TokenMint = {
+      // Mint 50 BTC
+      balances: [{ token: Number(idBTC), amount: new BigNumber(10) }],
+      to: {
+        stack: []
+      }
+    }
+
+    const txn = await bBuilder.tokens.mint(tokenMint, script)
+    const promise = sendTransaction(bob.container, txn)
+
+    await expect(promise).rejects.toThrow("MintTokenTx: You will exceed global daily maximum consortium mint limit for BTC token by minting this amount. (code 16)', code: -26")
+  })
+
+  it('should mintTokens', async () => {
+    await setupGovs()
+
+    const script = await aProviders.elliptic.script()
+    const tokenMint: TokenMint = {
+      // Mint 11.5 BTC
+      balances: [{ token: Number(idBTC), amount: new BigNumber(11.5) }],
+      to: {
+        stack: []
+      }
+    }
+
+    const txn = await aBuilder.tokens.mint(tokenMint, script)
+
+    // Ensure the created txn is correct
+    const outs = await sendTransaction(alice.container, txn)
+    const encoded: string = OP_CODES.OP_DEFI_TX_TOKEN_MINT(tokenMint).asBuffer().toString('hex')
+    const pubKey = await aProviders.ellipticPair.publicKey()
+    const address = Bech32.fromPubKey(pubKey, 'bcrt')
+
+    expect(outs).toStrictEqual([{
+      n: 0,
+      scriptPubKey: {
+        asm: expect.stringMatching(/^OP_RETURN 446654784d/),
+        hex: `6a${encoded}`,
+        type: 'nulldata'
+      },
+      tokenId: 0,
+      value: 0
+    }, {
+      n: 1,
+      scriptPubKey: {
+        addresses: [address],
+        asm: expect.any(String),
+        hex: expect.any(String),
+        reqSigs: 1,
+        type: 'witness_v0_keyhash'
+      },
+      tokenId: 0,
+      value: 99.9999929
+    }])
+
+    await alice.generate(1)
+
+    // Verify that the amount is minted to consortium member account
+    const accAfter = await alice.rpc.account.getAccount(consortiumMemberA.bech32)
+    expect(accAfter).toStrictEqual(['11.50000000@BTC'])
+
+    // Verify that the amount reflects to consortium member and token level attributes
+    const attr = (await alice.rpc.masternode.getGov(attributeKey)).ATTRIBUTES
+    expect(attr[`v0/live/economy/consortium_members/${idBTC}/01/minted`]).toStrictEqual(new BigNumber('11.5'))
+    expect(attr[`v0/live/economy/consortium_members/${idBTC}/01/daily_minted`]).toStrictEqual(`0/${new BigNumber(11.5).toFixed(8)}`)
+    expect(attr['v0/live/economy/consortium/1/minted']).toStrictEqual(new BigNumber('11.5'))
+  })
+
+  it('should allow to mint tokens to an address', async () => {
```

Comment:
Can we add a test where we mint a non supported token?

Reply:
Updated! https://github.com/JellyfishSDK/jellyfish/pull/2004/files#diff-937037dc208d4d941791f1ad7941f5a21f17481e07af61b5c2719531980f454fR178